runtime/ops/net.rs

```diff
@@ -528,3 +544,141 @@ fn op_listen(
     _ => Err(type_error("Wrong argument format!")),
   }
 }
+
+async fn op_resolve_addr(
+  state: Rc<RefCell<OpState>>,
+  args: Value,
+  _zero_copy: BufVec,
+) -> Result<Value, AnyError> {
+  fn default_port() -> u16 {
+    53
+  }
+
+  #[derive(Deserialize)]
+  struct ResolveAddrArgs {
+    query: String,
+    options: Option<ResolveAddrOption>,
+  }
+
+  #[derive(Deserialize)]
+  #[serde(rename_all = "camelCase")]
+  struct ResolveAddrOption {
+    #[serde(default)]
+    record_type: SupportedRecordType,
+    name_server: Option<NameServer>,
+  }
+
+  // TODO(magurotuna): There are many unsupported record types because `trust_dns_proto::rr::record_data::RData` doesn't implement `Display` trait.
+  // See https://github.com/bluejekyll/trust-dns/issues/897
+  #[derive(Deserialize, Clone, Copy)]
+  enum SupportedRecordType {
+    A,
+    AAAA,
+    ANAME,
+    CNAME,
+    PTR,
+  }
+
+  impl Default for SupportedRecordType {
+    fn default() -> Self {
+      Self::A
+    }
+  }
+
+  impl Into<RecordType> for SupportedRecordType {
+    fn into(self) -> RecordType {
+      use SupportedRecordType::*;
+      match self {
+        A => RecordType::A,
+        AAAA => RecordType::AAAA,
+        ANAME => RecordType::ANAME,
+        CNAME => RecordType::CNAME,
+        PTR => RecordType::PTR,
+      }
+    }
+  }
+
+  #[derive(Deserialize, Clone, Copy)]
+  enum Protocol {
+    UDP,
+    TCP,
+  }
+
+  impl Default for Protocol {
+    fn default() -> Self {
+      Self::UDP
+    }
+  }
+
+  impl Into<TrustProtocol> for Protocol {
+    fn into(self) -> TrustProtocol {
+      use Protocol::*;
+      match self {
+        UDP => TrustProtocol::Udp,
+        TCP => TrustProtocol::Tcp,
+      }
+    }
+  }
+
+  #[derive(Deserialize)]
+  #[serde(rename_all = "camelCase")]
+  struct NameServer {
+    ip_addr: String,
+    #[serde(default = "default_port")]
+    port: u16,
+    protocol: Protocol,
+  }
+
+  let ResolveAddrArgs { query, options } = serde_json::from_value(args)?;
+
+  {
+    let s = state.borrow();
+    // TODO(magurotuna): what permission to check?
+    s.borrow::<Permissions>().check_net("TODO", 0)?;
+  }
```

Comment:
I'm wondering what permission to check in this case. 
`check_net` requires hostname and port as arguments, but I think it would make sense to check just if `allow-net` is present, while neither hostname nor port is always necessary.

Reply:
You could check for resolver address + port 53. That might be difficult when you want to use the system resolver, though.

Reply:
We might also add another method to `Permissions` if that helps

Reply:
@mrkurt Thanks! That makes sense. So, if a user calls `Deno.resolveDns` with (a) name server address and/or (b) its port number specified, then we will check the permission like `check_net(a, b)`.